src/rebar_prv_shell.erl

```diff
@@ -134,6 +138,31 @@ setup_paths(State) ->
     %% add project app test paths
     ok = add_test_paths(State).
 
+maybe_run_script(State) ->
+    {Opts, _} = rebar_state:command_parsed_args(State),
+    case proplists:get_value(script_file, Opts, undefined) of
+        undefined ->
+            ?DEBUG("No script_file specified.", []),
+            ok;
+        File ->
+            run_script_file(File)
+    end.
+
+run_script_file(File) ->
+    ?DEBUG("Extracting escript from ~p", [File]),
+    {ok, Script} = escript:extract(File, [compile_source]),
+    Beam = proplists:get_value(source, Script),
+    Mod = proplists:get_value(module, beam_lib:info(Beam)),
+    ?DEBUG("Compiled escript as ~p", [Mod]),
+    FakeFile = "/fake_path/" ++ atom_to_list(Mod),
+    {module, Mod} = code:load_binary(Mod, FakeFile, Beam),
+    ?DEBUG("Loaded ~p", [Mod]),
+    erlang:group_leader(whereis(user), self()),
```

Comment:
The shell should be doing that in `setup_shell()` where it kills the old user, then starts a new one and registers it. I don't think this would be necessary here?